hack/perf-ibmcloud-openshift.sh

```diff
@@ -0,0 +1,250 @@
+    -nf|--node-flavor)        NODE_FLAVOR="$2";        shift;shift ;;
+    -n|--nodes)               NODES="$2";              shift;shift ;;
+    -ov|--openshift-version)  OPENSHIFT_VERSION="$2";  shift;shift ;;
+    -r|--region)              REGION="$2";             shift;shift ;;
+    -h|--help )
+      cat <<HELPMSG
+Create a performance cluster on IBM Cloud and install istio, kiali, and the testing demos.
+
+Usage:
+$0 [option...]
+
+Valid options:
+  -akf|--api-key-file
+    This is REQUIRED. A file containing the IBM Cloud api key used for creating the openshift cluster
+    and other resources in IBM Cloud. Learn how to create an API Key here:
+    https://www.ibm.com/docs/en/app-connect/containers_cd?topic=servers-creating-cloud-api-key.
+
+  -hr|--helm-charts-repo
+    The local dir where the helm charts repo is located.
+  
+  -iv|--istio-version
+    The Istio Version to install. Only upstream Istio supported.
+
+  -kv|--kiali-version
+    The Kiali version to install.
+
+  -nf|--node-flavor
+    The node flavor to use for the openshift cluster worker nodes.
+    Use 'ibmcloud ks flavors --zone <zone>' to see all available flavors in the region/zone.
+
+  -n|--nodes
+    The number of worker nodes in the openshift cluster's node pool.
+
+  -ov|--openshift-version
+    The version of openshift to install for the worker nodes.
+
+  -r|--region
+    The region to install the openshift worker nodes.
+
+  -h|--help )
+HELPMSG
+      exit 1
+      ;;
+    *)
+      echo "Unknown option: [$key]. Aborting. Use '$0 --help' for proper usage."
+      exit 1
+      ;;
+  esac
+done
+
+if [ ! -f ${API_KEY_FILE} ]; then
+  echo "The API Key file is invalid: ${API_KEY_FILE}"
+  exit 1
+fi
+
+# Set the config
+: "${HELM_CHARTS_REPO:=${DEFAULT_HELM_CHARTS_REPO}}"
+: "${ISTIO_VERSION:=${DEFAULT_ISTIO_VERSION}}"
+: "${KIALI_VERSION:=${DEFAULT_KIALI_VERSION}}"
+: "${NODE_FLAVOR:=${DEFAULT_NODE_FLAVOR}}"
+: "${NODES:=${DEFAULT_NODES}}"
+: "${OPENSHIFT_VERSION:=${DEFAULT_OPENSHIFT_VERSION}}"
+: "${REGION:=${DEFAULT_REGION}}"
+: "${API_KEY:-$(cat ${API_KEY_FILE})}"
+
+# Dump config
+infomsg "==START CONFIG=="
+cat<<EOM
+API_KEY_FILE=${API_KEY_FILE}
+HELM_CHARTS_REPO=${HELM_CHARTS_REPO}
+ISTIO_VERSION=${ISTIO_VERSION}
+KIALI_VERSION=${KIALI_VERSION}
+NODE_FLAVOR=${NODE_FLAVOR}
+NODES=${NODES}
+OPENSHIFT_VERSION=${OPENSHIFT_VERSION}
+REGION=${REGION}
+EOM
+infomsg "==END CONFIG=="
+
+install_kiali() {
+  infomsg "Installing kiali operator and kiali"
+
+  # cluster-status will expose the cluster's internal image registry route that podman needs to push images to.
+  make -C "${SCRIPT_DIR}/.." -e DORP=podman cluster-status
+  local cluster_repo
+  cluster_repo=$(oc get image.config.openshift.io/cluster -o custom-columns=EXT:.status.externalRegistryHostnames[0] --no-headers 2>/dev/null)
+  local master_url
+  master_url=$(ibmcloud oc cluster get -c kiali-perf-cluster --output json | jq -r '.masterURL')
```

Comment:
I think we should do this so we don't have N clusters called the same thing created by a bunch of different devs
```suggestion
  master_url=$(ibmcloud oc cluster get -c ${USER}-kiali-perf-cluster --output json | jq -r '.masterURL')
```